### PR TITLE
Implement the possibility to waive missing requirements via bodhi-cli

### PR DESCRIPTION
**Pull request by pypingou**
This commit basically implements the command: bodhi update waive, see
--help for the arguments.
It queries greenwave via bodhi to know the list of unsatisfied
requirements and uses the waiving endpoint in bodhi's API to waive either
all or just some of them.

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

### CODE REVIEW

File: bodhi/client/__init__.py [hunk 1]
```diff
@@ -621,6 +621,79 @@ def _get_notes(**kwargs):
...
+                       True.
+        kwargs (dict): Other keyword arguments passed to us by click.
+    """
+    client = bindings.BodhiClient(base_url=url, staging=kwargs['staging'])
+
+    if show and test:
+        click.echo(
+            'You can not list the unsatisfied requirements and waive them at the same '
+            'time, please use either --show or --test=... but not both.')
+        return
```
Comment (cverna) [inline bodhi/client/__init__.py]: other command are using `sys.exit(1)` in that case, we could try to stay consistent

File: bodhi/client/__init__.py [hunk 2]
```diff
@@ -621,6 +621,79 @@ def _get_notes(**kwargs):
...
+            if test_status.decision.unsatisfied_requirements:
+                click.echo('Missing tests:')
+                for req in test_status.decision.unsatisfied_requirements:
+                    click.echo('  - %s' % req)
+            else:
+                click.echo('Missing tests: None')
+    else:
+        if not comment:
+            click.echo('Comment are mandatory when waiving unsatisfied requirements')
+            return
```
Comment (cverna) [inline bodhi/client/__init__.py]: same here, we could use `sys.exit(1)` and add the `ERROR:` in front of the message

File: bodhi/client/__init__.py (hunk 1 above)
Comment (cverna) [reply bodhi/client/__init__.py]: also it seems that `ERROR:` is added at the beginning of the message just like here https://github.com/fedora-infra/bodhi/blob/develop/bodhi/client/__init__.py#L576

Comment (pypingou) [reply bodhi/client/__init__.py]: I'll adjust

File: bodhi/client/__init__.py (hunk 2 above)
Comment (pypingou) [reply bodhi/client/__init__.py]: Will do :)

**Comment by cverna** [comment]
:+1:

**Comment by bowlofeggs** [comment]
This patch is planned to be included in the upcoming 3.9.0 release: #2477.

**Comment by bowlofeggs** [comment]
This patch was released with 3.9.0 last week.
